app/controllers/application_controller/buttons.rb

```diff
@@ -676,10 +667,10 @@ def group_set_form_vars
         @edit[:new][:fields].push([mem.name,mem.id])
       end
     end
-    uri = CustomButton.buttons_for(@sb[:applies_to_class]).sort_by(&:name)
-    uri.each do |u|
-      @edit[:new][:available_fields].push([u.name,u.id]) if u.parent.nil?
-    end
+    @edit[:new][:available_fields] = CustomButton.buttons_for(@sb[:applies_to_class])
+                                                 .sort_by(&:name)
+                                                 .select  { |u| u.parent.nil? }
+                                                 .collect { |u| [u.name, u.id] }
```

Comment:
Might be better to sort *after* you filter with select.

Also, is parent a relation?  If so, then this is an N+1.  Is there a way to preload it with `includes`?  Since you are filtering, perhaps you can also eliminate them at the query level earlier.

(Note, I know this is a refactoring PR, so any query-change suggestions I have can wait until a followup PR)